### ISSUE TITLE
Handlebars partials are not supported

### ISSUE THREAD

**Issue by pdehaan** [ember-template-lint/ember-template-lint#486]
I'm invoking **ember-template-lint** via **npx** (to lint Handlebars *.hbs files) and I am getting a handful of the following error, but the rule name seems to be `undefined`, so I'm unsure how/if I can disable those errors:

```bash
$ npx ember-template-lint views/**/*.hbs

./views/confirm.hbs
  10:4  error  Handlebars partials are not supported: "{{> social_media_sharing_buttons" at L10:C4  undefined

./views/layouts/default.hbs
  27:6  error  Handlebars partials are not supported: "{{> subscribe_modal/subscribe_modal_wrapper" at L27:C6  undefined

...
```

And I have a .template-lintrc.js file that looks like this:

```js
module.exports = {
  extends: 'recommended',

  rules: {
    'block-indentation': false,
    'self-closing-void-elements': false
  }
};
```

**Comment by pdehaan** [comment]
Closest I can find is [this reference in Glimmer](https://github.com/glimmerjs/glimmer-vm/blob/25f84bdd9e989449ba794ea4432f14f05c1ff454/packages/%40glimmer/syntax/lib/parser/handlebars-node-visitors.ts#L206-L215):

```js
  PartialStatement(partial: HandlebarsAST.PartialStatement) {
    let { loc } = partial;

    throw new SyntaxError(
      `Handlebars partials are not supported: "${this.sourceForNode(partial, partial.name)}" at L${
        loc.start.line
      }:C${loc.start.column}`,
      partial.loc
    );
  }
```

**Comment by rwjblue** [comment]
Yep, that’s where the error is coming from. We use the glimmer parser (which is what ember uses) to do this linting.

What do your source templates look like?

**Comment by pdehaan** [comment]
@rwjblue Pretty boring:

```hbs
{{!< default }}
<div id="confirmation" class="sub-page-content">
  <div class="section-wrapper centered-content">
    <h3 class="section-headline whole">
      Confirmed!<br />
      You're Subscribed
    </h3>
    <p class="whole sub-head">Firefox Monitor will email you a full report shortly, and will send an email alert if your account appears in a new reported breach.</p>
    <h4 class="whole medium secondary-title">If your accounts have been compromised, chances are your friends, family, or online connections are too. Let them know about Firefox Monitor.</h4>
    {{> social_media_sharing_buttons}}
  </div>
</div>
```

**Comment by rwjblue** [comment]
The error in glimmer-vm when it sees partials is primarily trying to prevent that syntax from use in Ember apps. Unfortunately, it _also_ means that we cannot even parse the files for use in the template linter (since this parsing happens super early in the linting process, and is actually **how** linting works).

In order to make this work, we would change glimmer-vm to not throw an error at parse time but instead to throw at compilation time. That doesn't too outrageous to do, and I'm happy help guide someone through that process if they are interested...

**Comment by matijaboban** [comment]
hi @rwjblue just wondering is there any update on this issue?

Thanks

**Comment by rwjblue** [comment]
TBH, I think it may work now (after #856). You'd have to test...

**Comment by matijaboban** [comment]
@rwjblue Unfortunately did not do the trick (just tested with 1.7.0).

Still getting the "Handlebars partials are not supported" error.

Example code:
```
...
<td class="wrapper">
            <h1 class="align-center">Title</h1>

            <!-- Contains a partial -->
            {{> _example_partial }}

            <p>Content</p>
            <p>{{ _example_vars.dynamic_content }}</p>

            <!-- Button component, same as a partial -->
            {{> button type='primary' align='center' url='#' title='Demo Button' }}

            <!-- Handlebars escaped for Mailgun -->
            {{> mg_if var='mailgun_var' }}
        </td>
...
```

LInting error returned:
```
{...}/src/emails/_example_email.hbs
  15:12  error  Handlebars partials are not supported: "{{> _example_partial" at L15:C12  undefined
...
```

**Comment by iwaduarte** [comment]
Any direction on that?

**Comment by lifeart** [comment]
ember-template-lint hardcoded to use parser from https://github.com/glimmerjs/glimmer-vm
and glimmer parser does not support partials

closing issue as wontfix
